src/language-services/analysis.ts

```diff
@@ -15,53 +19,50 @@ import {
     SignatureProviderContext,
     SymbolProvider,
 } from "./providers";
+import * as WorkspaceCache from "./workspaceCache";
 
-export interface Analysis {
+export interface Analysis extends IDisposable {
     getCompletionItems(): Promise<CompletionItem[]>;
     getHover(): Promise<Hover>;
     getSignatureHelp(): Promise<SignatureHelp>;
 }
 
-export interface AnalysisOptions {
-    readonly environmentSymbolProvider?: SymbolProvider;
-    readonly librarySymbolProvider?: LibrarySymbolProvider;
-}
-
-export function createAnalysisSession(document: TextDocument, position: Position, options: AnalysisOptions): Analysis {
+export function createAnalysisSession(document: TextDocument, position: Position, options: Configuration): Analysis {
     return new DocumentAnalysis(document, position, options);
 }
 
-class DocumentAnalysis implements Analysis {
-    private readonly maybeTriedInspection: PQP.Task.TriedInspection | undefined;
-    private readonly environmentSymbolProvider: SymbolProvider;
-    private readonly keywordProvider: KeywordProvider;
-    private readonly librarySymbolProvider: LibrarySymbolProvider;
-    private readonly localSymbolProvider: SymbolProvider;
-
-    constructor(
-        private readonly document: TextDocument,
-        private readonly position: Position,
-        options: AnalysisOptions,
-    ) {
-        this.maybeTriedInspection = WorkspaceCache.maybeTriedInspection(this.document, this.position);
-
-        this.environmentSymbolProvider = options.environmentSymbolProvider
-            ? options.environmentSymbolProvider
+abstract class AnalysisBase implements Analysis {
+    protected readonly environmentSymbolProvider: SymbolProvider;
+    protected readonly keywordProvider: KeywordProvider;
+    protected readonly librarySymbolProvider: LibrarySymbolProvider;
+    protected readonly localSymbolProvider: SymbolProvider;
+
+    protected readonly options: Configuration;
+    protected readonly position: Position;
+    protected readonly triedInspection: PQP.Task.TriedInspection | undefined;
```

Comment:
At what level are we dropping the maybe/try prefix on var names? There's a lot more maybe/try remaining in this repo.

Reply:
I was trying to do it on anything we export. I believe we'll need to move away from it entirely if want to use the same naming rules from the Power Query codebase.